Adding `rustls` killed installing & upgrading Nu with all features

### Describe the bug

https://github.com/nushell/nushell/pull/15810 introduced this bug.

```
cargo install --locked --all-features nu
    Updating crates.io index
  Installing nu v0.105.1
    Updating crates.io index
    Updating crates.io index
   Compiling proc-macro2 v1.0.92
   Compiling unicode-ident v1.0.14
   Compiling cfg-if v1.0.0
   Compiling libc v0.2.168
   Compiling serde v1.0.216
   Compiling memchr v2.7.4
   Compiling autocfg v1.4.0
   Compiling rustix v0.38.42
   Compiling linux-raw-sys v0.4.14
   Compiling log v0.4.22
   Compiling once_cell v1.20.2
   Compiling foldhash v0.1.5
   Compiling allocator-api2 v0.2.21
   Compiling equivalent v1.0.1
   Compiling iana-time-zone v0.1.61
   Compiling pure-rust-locales v0.8.1
   Compiling smallvec v1.13.2
   Compiling utf8parse v0.2.2
   Compiling itoa v1.0.14
   Compiling thiserror v2.0.12
   Compiling byteorder v1.5.0
   Compiling unicode-width v0.1.11
   Compiling cfg_aliases v0.2.1
   Compiling ryu v1.0.18
   Compiling shlex v1.3.0
   Compiling version_check v0.9.5
   Compiling crossbeam-utils v0.8.21
   Compiling thiserror v1.0.69
   Compiling either v1.15.0
   Compiling adler2 v2.0.0
   Compiling serde_json v1.0.133
   Compiling arrayvec v0.7.6
   Compiling scopeguard v1.2.0
   Compiling parking_lot_core v0.9.10
   Compiling miniz_oxide v0.8.0
   Compiling regex-syntax v0.8.5
   Compiling crc32fast v1.4.2
   Compiling nix v0.29.0
   Compiling signal-hook v0.3.17
   Compiling rustversion v1.0.18
   Compiling pkg-config v0.3.31
   Compiling typenum v1.17.0
   Compiling cc v1.2.16
   Compiling paste v1.0.15
   Compiling heck v0.5.0
   Compiling nu-ansi-term v0.50.1
   Compiling rayon-core v1.12.1
   Compiling bit-vec v0.8.0
   Compiling typeid v1.0.2
   Compiling gimli v0.31.1
   Compiling option-ext v0.2.0
   Compiling generic-array v0.14.7
   Compiling rustc-demangle v0.1.24
   Compiling procfs v0.17.0
   Compiling hashbrown v0.15.2
   Compiling unicase v2.8.0
   Compiling num-traits v0.2.19
   Compiling lock_api v0.4.12
   Compiling aho-corasick v1.1.3
   Compiling flate2 v1.0.35
   Compiling object v0.36.5
   Compiling bit-set v0.8.0
   Compiling owo-colors v4.1.0
   Compiling ref-cast v1.0.23
   Compiling hex v0.4.3
   Compiling alloc-no-stdlib v2.0.4
   Compiling num-format v0.4.4
   Compiling sys-locale v0.3.2
   Compiling is_ci v1.2.0
   Compiling unicode-linebreak v0.1.5
   Compiling itertools v0.14.0
   Compiling siphasher v0.3.11
   Compiling alloc-stdlib v0.2.2
   Compiling rand_core v0.6.4
   Compiling supports-color v3.0.2
   Compiling inventory v0.3.15
   Compiling brotli-decompressor v4.0.1
   Compiling supports-unicode v3.0.0
   Compiling bytes v1.10.0
   Compiling textwrap v0.16.1
   Compiling supports-hyperlinks v3.1.0
   Compiling web-time v1.1.0
   Compiling phf_shared v0.11.2
   Compiling rand v0.8.5
   Compiling nu-glob v0.105.1
   Compiling minimal-lexical v0.2.1
   Compiling semver v1.0.23
   Compiling indexmap v2.9.0
   Compiling lru v0.12.5
   Compiling vcpkg v0.2.15
   Compiling crossbeam-epoch v0.9.18
   Compiling quote v1.0.37
   Compiling nom v7.1.3
   Compiling same-file v1.0.6
   Compiling stable_deref_trait v1.2.0
   Compiling getrandom v0.3.1
   Compiling walkdir v2.5.0
   Compiling doctest-file v1.0.0
   Compiling syn v2.0.90
   Compiling unicode-width v0.2.0
   Compiling wayland-sys v0.31.5
   Compiling crossbeam-deque v0.8.5
   Compiling anstyle-parse v0.2.6
   Compiling vte_generate_state_changes v0.1.2
   Compiling proc-macro-error-attr2 v2.0.0
   Compiling phf_generator v0.11.2
   Compiling rmp v0.8.14
   Compiling libloading v0.8.6
   Compiling phf_codegen v0.11.2
   Compiling anstyle-query v1.1.2
   Compiling litemap v0.7.4
   Compiling anstyle v1.0.10
   Compiling is_terminal_polyfill v1.70.1
   Compiling colorchoice v1.0.3
   Compiling writeable v0.5.5
   Compiling phf v0.11.2
   Compiling clap_lex v0.7.4
   Compiling fnv v1.0.7
   Compiling strsim v0.11.1
   Compiling dlib v0.5.2
   Compiling quick-xml v0.36.2
   Compiling vte v0.11.1
   Compiling openssl-src v300.4.1+3.4.0
   Compiling number_prefix v0.4.0
   Compiling rustix v1.0.7
   Compiling unicode-xid v0.2.6
   Compiling wayland-client v0.31.7
   Compiling brotli v7.0.0
   Compiling anstream v0.6.18
   Compiling uuhelp_parser v0.0.30
   Compiling bytesize v1.3.3
   Compiling lscolors v0.20.0
   Compiling wayland-backend v0.3.7
   Compiling downcast-rs v1.2.1
   Compiling icu_locid_transform_data v1.5.0
   Compiling scoped-tls v1.0.1
   Compiling const_format_proc_macros v0.2.34
   Compiling ahash v0.8.11
   Compiling os_display v0.1.3
   Compiling percent-encoding v2.3.1
   Compiling glob v0.3.1
   Compiling icu_properties_data v1.5.0
   Compiling cfg_aliases v0.1.1
   Compiling openssl-sys v0.9.107
   Compiling uucore_procs v0.0.30
   Compiling dunce v1.0.5
   Compiling unicode-segmentation v1.12.0
   Compiling rayon v1.10.0
   Compiling wild v2.2.1
   Compiling linux-raw-sys v0.9.4
   Compiling nix v0.28.0
   Compiling ring v0.17.13
   Compiling num-conv v0.1.0
   Compiling regex-automata v0.4.9
   Compiling utf16_iter v1.0.5
   Compiling rustls-pki-types v1.10.0
   Compiling fastrand v2.3.0
   Compiling write16 v1.0.0
   Compiling bitflags v1.3.2
   Compiling utf8_iter v1.0.4
   Compiling fixedbitset v0.4.2
   Compiling icu_normalizer_data v1.5.0
   Compiling signal-hook-registry v1.4.2
   Compiling mio v1.0.3
   Compiling strip-ansi-escapes v0.2.0
   Compiling os_pipe v1.2.1
   Compiling dirs-sys v0.4.1
   Compiling interprocess v2.2.2
   Compiling block-buffer v0.10.4
   Compiling crypto-common v0.1.6
   Compiling wayland-scanner v0.31.5
   Compiling parking_lot v0.12.3
   Compiling digest v0.10.7
   Compiling dirs v5.0.1
   Compiling time-core v0.1.2
   Compiling lazy_static v1.5.0
   Compiling addr2line v0.24.2
   Compiling petgraph v0.6.5
   Compiling form_urlencoded v1.2.1
   Compiling libsqlite3-sys v0.28.0
   Compiling vte v0.14.1
   Compiling itertools v0.13.0
   Compiling encoding_rs v0.8.35
   Compiling signal-hook-mio v0.2.4
   Compiling ucd-trie v0.1.7
   Compiling getrandom v0.2.15
   Compiling tz-rs v0.7.0
   Compiling ansitok v0.3.0
   Compiling powerfmt v0.2.0
   Compiling ident_case v1.0.1
   Compiling syn v1.0.109
   Compiling home v0.5.9
   Compiling bytemuck v1.23.0
   Compiling portable-atomic v1.10.0
   Compiling openssl v0.10.72
   Compiling foreign-types-shared v0.1.1
   Compiling untrusted v0.9.0
   Compiling x11rb-protocol v0.13.1
   Compiling deranged v0.3.11
   Compiling foreign-types v0.3.2
   Compiling md-5 v0.10.6
   Compiling console v0.15.8
   Compiling filetime v0.2.25
   Compiling crossbeam-channel v0.5.15
   Compiling atomic v0.6.0
   Compiling slab v0.4.9
   Compiling native-tls v0.2.12
   Compiling cpufeatures v0.2.16
   Compiling openssl-probe v0.1.5
   Compiling fallible-iterator v0.3.0
   Compiling fallible-streaming-iterator v0.1.9
   Compiling num_threads v0.1.7
   Compiling tzdb_data v0.2.1
   Compiling ansi-str v0.9.0
   Compiling rustls v0.23.20
   Compiling sha1_smol v1.0.1
   Compiling zerocopy v0.8.23
   Compiling bytecount v0.6.8
   Compiling uuid v1.16.0
   Compiling sha2 v0.10.8
   Compiling papergrid v0.17.0
   Compiling const_format v0.2.34
   Compiling rand_core v0.9.3
   Compiling inotify-sys v0.1.5
   Compiling time v0.3.37
   Compiling rust_decimal v1.36.0
   Compiling subtle v2.6.1
   Compiling zeroize v1.8.1
   Compiling mime v0.3.17
   Compiling lockfree-object-pool v0.1.6
   Compiling v_htmlescape v0.15.8
   Compiling regex v1.11.1
   Compiling zip v2.4.2
   Compiling arrayvec v0.5.2
   Compiling bumpalo v3.16.0
   Compiling simd-adler32 v0.3.7
   Compiling is_debug v1.1.0
   Compiling indicatif v0.17.9
   Compiling inotify v0.9.6
   Compiling rust-embed-utils v8.7.0
   Compiling mime_guess v2.0.5
   Compiling vte v0.10.1
   Compiling zopfli v0.8.1
   Compiling webpki-roots v0.26.8
   Compiling parse-zoneinfo v0.3.1
   Compiling socks v0.3.4
   Compiling sysinfo v0.33.1
   Compiling tree_magic_mini v3.1.6
   Compiling mio v0.8.11
   Compiling fluent-uri v0.1.4
   Compiling castaway v0.2.3
   Compiling backtrace v0.3.74
   Compiling nom v8.0.0
   Compiling nu-pretty-hex v0.105.1
   Compiling thread_local v1.1.8
   Compiling is-docker v0.2.0
   Compiling indoc v2.0.5
   Compiling static_assertions v1.1.0
   Compiling base64 v0.22.1
   Compiling chrono-tz-build v0.4.0
   Compiling futures-task v0.3.31
   Compiling futures-core v0.3.31
   Compiling winnow v0.6.20
   Compiling anyhow v1.0.94
   Compiling pin-utils v0.1.0
   Compiling pin-project-lite v0.2.15
   Compiling ansitok v0.2.0
   Compiling compact_str v0.8.0
   Compiling is-wsl v0.4.0
   Compiling notify v6.1.1
   Compiling tabled v0.20.0
   Compiling backtrace-ext v0.2.1
   Compiling fuzzy-matcher v0.3.7
   Compiling tzdb v0.7.2
   Compiling synstructure v0.13.1
   Compiling proc-macro-error2 v2.0.1
   Compiling chrono-tz v0.10.0
   Compiling darling_core v0.20.10
   Compiling unicode-truncate v1.1.0
   Compiling quick-xml v0.31.0
   Compiling codepage v0.1.2
   Compiling fancy-regex v0.14.0
   Compiling etcetera v0.8.0
   Compiling pori v0.0.0
   Compiling time-macros v0.2.19
   Compiling nucleo-matcher v0.3.1
   Compiling platform-info v2.0.4
   Compiling csv-core v0.1.11
   Compiling itertools v0.11.0
   Compiling shadow-rs v1.1.1
   Compiling is_executable v1.0.4
   Compiling cassowary v0.3.0
   Compiling unsafe-libyaml v0.2.11
   Compiling urlencoding v2.1.3
   Compiling shell-words v1.1.0
   Compiling fs_extra v1.3.0
   Compiling pathdiff v0.2.3
   Compiling quick-error v2.0.1
   Compiling file-id v0.2.2
   Compiling open v5.3.1
   Compiling notify-debouncer-full v0.3.2
   Compiling ansi-str v0.8.0
   Compiling chardetng v0.1.17
   Compiling rustls-native-certs v0.8.1
   Compiling devicons v0.6.12
   Compiling webpki-roots v1.0.0
   Compiling nu-cmd-lang v0.105.1
   Compiling print-positions v0.6.1
   Compiling quick-xml v0.37.1
   Compiling nu v0.105.1
   Compiling alphanumeric-sort v1.5.3
   Compiling data-encoding v2.9.0
   Compiling bracoxide v0.1.6
   Compiling termcolor v1.4.1
   Compiling serde_derive v1.0.216
   Compiling thiserror-impl v2.0.12
   Compiling thiserror-impl v1.0.69
   Compiling strum_macros v0.26.4
   Compiling ref-cast-impl v1.0.23
   Compiling miette-derive v7.6.0
   Compiling typetag-impl v0.2.18
   Compiling nu-derive-value v0.105.1
   Compiling zerofrom-derive v0.1.5
   Compiling yoke-derive v0.7.5
   Compiling displaydoc v0.2.5
   Compiling zerovec-derive v0.10.3
   Compiling zerocopy-derive v0.7.35
   Compiling icu_provider_macros v1.5.0
   Compiling derive-new v0.6.0
   Compiling openssl-macros v0.1.1
   Compiling pest_consume_macros v1.1.0
   Compiling serde_repr v0.1.19
   Compiling futures-macro v0.3.31
   Compiling rust-embed-impl v8.7.0
   Compiling filesize v0.2.0
   Compiling roxmltree v0.20.0
   Compiling darling_macro v0.20.10
   Compiling zerocopy v0.7.35
   Compiling rust-embed v8.7.0
   Compiling pwd v1.4.0
   Compiling umask v2.1.0
   Compiling dialoguer v0.11.0
   Compiling wax v0.6.0
   Compiling darling v0.20.10
   Compiling futures-util v0.3.31
   Compiling pest v2.7.15
   Compiling instability v0.3.7
   Compiling nu-path v0.105.1
   Compiling zerofrom v0.1.5
   Compiling yoke v0.7.5
   Compiling zerovec v0.10.4
   Compiling strum v0.26.3
   Compiling simplelog v0.12.2
   Compiling ppv-lite86 v0.2.20
   Compiling rustls-webpki v0.102.8
   Compiling hashbrown v0.14.5
   Compiling rand_chacha v0.9.0
   Compiling rand v0.9.0
   Compiling titlecase v3.5.0
   Compiling pest_meta v2.7.15
   Compiling tinystr v0.7.6
   Compiling icu_collections v1.5.0
   Compiling hashlink v0.9.1
   Compiling icu_locid v1.5.0
   Compiling pest_generator v2.7.15
   Compiling multipart-rs v0.1.13
   Compiling icu_provider v1.5.0
   Compiling icu_locid_transform v1.5.0
   Compiling pest_derive v2.7.15
   Compiling pest_consume v1.1.3
   Compiling icu_properties v1.5.1
   Compiling bitflags v2.6.0
   Compiling chrono v0.4.39
   Compiling erased-serde v0.4.5
   Compiling rmp-serde v1.3.0
   Compiling linked-hash-map v0.5.6
   Compiling toml_datetime v0.6.8
   Compiling serde_spanned v0.6.8
   Compiling serde_urlencoded v0.7.1
   Compiling csv v1.3.1
   Compiling serde_yaml v0.9.34+deprecated
   Compiling toml_edit v0.22.22
   Compiling typetag v0.2.18
   Compiling nu-json v0.105.1
   Compiling lsp-types v0.97.0
   Compiling lsp-server v0.7.8
   Compiling oem_cp v2.0.0
   Compiling icu_normalizer v1.5.0
   Compiling procfs-core v0.17.0
   Compiling chrono-humanize v0.2.3
   Compiling idna_adapter v1.2.0
   Compiling parse_datetime v0.8.0
   Compiling trash v5.2.1
   Compiling human-date-parser v0.3.1
   Compiling calamine v0.26.1
   Compiling idna v1.0.3
   Compiling dtparse v2.0.1
   Compiling url v2.5.4
   Compiling ctrlc v3.4.5
   Compiling tempfile v3.20.0
   Compiling toml v0.8.19
   Compiling terminal_size v0.4.1
   Compiling crossterm v0.28.1
   Compiling xattr v1.3.1
   Compiling miette v7.6.0
   Compiling clap_builder v4.5.23
   Compiling x11rb v0.13.1
   Compiling fd-lock v4.0.2
   Compiling which v7.0.0
   Compiling nu-system v0.105.1
   Compiling nu-utils v0.105.1
   Compiling ratatui v0.29.0
   Compiling nu-protocol v0.105.1
   Compiling nu-term-grid v0.105.1
   Compiling clap v4.5.23
   Compiling uucore v0.0.30
   Compiling wayland-protocols v0.31.2
   Compiling uu_touch v0.0.30
   Compiling uu_whoami v0.0.30
   Compiling uu_mkdir v0.0.30
   Compiling uu_mktemp v0.0.30
   Compiling uu_cp v0.0.30
   Compiling uu_mv v0.0.30
   Compiling uu_uname v0.0.30
   Compiling wayland-protocols-wlr v0.2.0
   Compiling wl-clipboard-rs v0.8.1
   Compiling arboard v3.4.1
   Compiling lsp-textdocument v0.4.2
   Compiling nu-engine v0.105.1
   Compiling nu-plugin-protocol v0.105.1
   Compiling nu-plugin-core v0.105.1
   Compiling nu-color-config v0.105.1
   Compiling nu-plugin-engine v0.105.1
   Compiling nu-table v0.105.1
   Compiling nu-parser v0.105.1
   Compiling nu-cmd-plugin v0.105.1
   Compiling ureq v2.12.1
   Compiling update-informer v1.2.0
   Compiling nu-cmd-base v0.105.1
   Compiling nuon v0.105.1
   Compiling nu-explore v0.105.1
   Compiling nu-std v0.105.1
   Compiling nu-cmd-extra v0.105.1
   Compiling rusqlite v0.31.0
   Compiling reedline v0.40.0
   Compiling nu-cli v0.105.1
   Compiling nu-command v0.105.1
error[E0428]: the name `impl_tls` is defined multiple times
  --> /home/user/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/nu-command-0.105.1/src/network/tls/mod.rs:14:1
   |
10 | mod impl_tls;
   | ------------- previous definition of the module `impl_tls` here
...
14 | mod impl_tls;
   | ^^^^^^^^^^^^^ `impl_tls` redefined here
   |
   = note: `impl_tls` must be defined only once in the type namespace of this module

error: Multiple TLS backends enabled. Please enable only one of `native-tls` or `rustls-tls`, not both.
  --> /home/user/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/nu-command-0.105.1/src/network/tls/mod.rs:22:1 
   |
22 | / compile_error!(
23 | |     "Multiple TLS backends enabled. Please enable only one of `native-tls` or `rustls-tls`, not both."
24 | | );
   | |_^

   Compiling nu-lsp v0.105.1
For more information about this error, try `rustc --explain E0428`.
error: could not compile `nu-command` (lib) due to 2 previous errors
error: failed to compile `nu v0.105.1`, intermediate artifacts can be found at `/tmp/cargo-installHG5hW6`.
To reuse those artifacts with a future compilation, set the environment variable `CARGO_TARGET_DIR` to that path.
```

The ideal solution would be to fix `--all-features`, so it includes only the default TLS implementation. Otherwise, the flag `--all-features` will remain borked in general.

### How to reproduce

cargo install --locked --all-features nu

### Expected behavior

To upgrade without issues, as usual.

### Configuration

| key                | value                                                         |
| ------------------ | ------------------------------------------------------------- |
| version            | 0.104.1                                                       |
| major              | 0                                                             |
| minor              | 104                                                           |
| patch              | 1                                                             |
| branch             |                                                               |
| commit_hash        |                                                               |
| build_os           | linux-x86_64                                                  |
| build_target       | x86_64-unknown-linux-gnu                                      |
| rust_version       | rustc 1.86.0 (05f9846f8 2025-03-31)                           |
| rust_channel       | stable-x86_64-unknown-linux-gnu                               |
| cargo_version      | cargo 1.86.0 (adf9b6ad1 2025-02-28)                           |
| build_rust_channel | release                                                       |
| allocator          | standard                                                      |
| features           | default, sqlite, static-link-openssl, system-clipboard, trash |
| installed_plugins  |                                                               |

--- a/Cargo.toml
+++ b/Cargo.toml
@@ -257,6 +257,10 @@ serial_test = "3.2"
 tempfile = { workspace = true }
 
 [features]
+# Enable all features while still avoiding mutually exclusive features.
+# Use this if `--all-features` fails.
+full = ["plugin", "rustls-tls", "system-clipboard", "trash-support", "sqlite"]
+
 plugin = [
   # crates
   "nu-cmd-plugin",

